client/python/gradio_client/client.py

```diff
@@ -1104,6 +1118,66 @@ def _inner(*data):
 
         return _inner
 
+    def make_cancel(
+        self,
+    ):
+        if self.client.app_version > version.Version("4.29.0"):
+            url = urllib.parse.urljoin(self.client.src, utils.CANCEL_URL)
+            data = {
+                "fn_index": self.fn_index,
+                "session_hash": self.client.session_hash,
+            }
+            cancel_msg = None
+            cancellable = True
+        else:
+            candidates: list[tuple[int, list[int]]] = []
+            for i, dep in enumerate(self.client.config["dependencies"]):
+                if self.fn_index in dep["cancels"]:
+                    candidates.append(
+                        (i, [d for d in dep["cancels"] if d != self.fn_index])
+                    )
+
+            fn_index, other_cancelled = (
+                min(candidates, key=lambda x: len(x[1])) if candidates else (None, None)
+            )
+            cancel_msg = None
+            cancellable = fn_index is not None
+            cancel_msg = (
+                "Cancelling this job will not stop the server from running. "
```

Comment:
Verified this warning is triggered correctly

<img width="906" alt="image" src="https://github.com/gradio-app/gradio/assets/41651716/dd504379-2812-4d4f-9bf9-c47815d399e8">

Reply:
In my testing, I noticed that this message appears even if the Gradio app (is running 4.29 and) already includes an event that cancels this event